Feature customer 2

ksereda opened the pull request.

Dimian13 committed:
Merge remote-tracking branch 'origin/development' into feature-customer_2

Dimian13 committed:
Merge remote-tracking branch 'origin/development' into feature-customer_2

Dimian13 committed:
added description of DAO level

ksereda committed:
add Service to readme.md file

ksereda committed:
add description Service to README.md file

Dimian13 committed:
added client for belavia soap service to project

Dimian13 committed:
added date validation, refactored methods

Dimian13 committed:
added verification of language parameter to "RU" or "EN" values. Added exceptions catching to AppController class. Added exceptions messegges to modelAndView. Added colors and aligne of exception messegges to jsp pages.

ksereda committed:
begin write soap service for convert celsioum to fahrenheit

ksereda committed:
delete my soap service

Dimian13 committed:
Merge branch 'development' into feature-customer_2

# Conflicts:
#	src/main/java/io/delivery/config/AppConfig.java
#	src/main/java/io/delivery/controller/AppController.java

Dimian13 committed:
renamed Client.java from belavia/webservices to ClientBelavia.java
ClientBelavia added to AppConfig as a @Bean
changed methods in AppController from Client to ClientBelavia
added imports of ClientBelavia to AppConfig and AppController

IvanSeregin commented on src/main/java/io/delivery/by/belavia/webservices/OnlineTimeTableSoap.java:
```diff
@@ -0,0 +1,63 @@
+
+
+    /**
+     * <h4>���������� ������ ������ � ����������� � ������ / ������ � ��������� �����.</h4>������� ���������:<ul><li>Airport � IATA ��� ���������, ������ �� ������ ��������� ����������</li><li>Type � ��� ���������� ��� ������, ����� ��������� �������� Arrival ����������� ������� �� ���������� ���������, Departure - ���������� ������� � ��������� ��������.</li><li>ViewDate � ���� � ������� (YYYY-M ...
```
Проблемы с кодировкой?

dbodunov replied:
Да, но в сгенерированных соап сервисом классах.

IvanSeregin commented on src/main/java/io/delivery/by/belavia/webservices/OnlineTimeTableSoap.java:
```diff
@@ -0,0 +1,63 @@
+        XMLGregorianCalendar viewDate);
+
+    /**
+     * <h4>���������� ������ ����������, ��� ���� ���������� � ����������� ������.</h4>������� ���������:<ul><li>Language - ������ ��, ����� ����� ������ ����� �������� ������. �� ������ ������ ���������� �������� ��������� �EN�, �RU�. ������� ����������� ���������� �� �����</li></ul><br />������������ ������, ��������� �� ��������� �����:<ul><li>IATA -  ��� IATA, ���������</li><li>Name � ������� ...
```
Проблемы с кодировкой?

dbodunov replied:
аналогично

IvanSeregin commented on src/main/java/io/delivery/by/belavia/webservices/ClientBelavia.java:
```diff
@@ -0,0 +1,137 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class ClientBelavia {
```
Как по мне, Belavia как-то не очень, пытался расшифровать и только спустя некоторое время понял, что это Беларусские авиалинии, может ClientAirlines как-нибудь или типа того.

dbodunov replied:
Название изменено на ClientBelaviaAirlines

IvanSeregin commented on src/main/java/io/delivery/controller/AppController.java:
```diff
@@ -78,15 +81,51 @@ public String noregisration() {
+        return modelAndView;
+    }
+
+    @RequestMapping(value = {"/belavia/timetable/{airport}/{type}/{date}"}, method = RequestMethod.GET)
```
Как по мне, лучше маппинги вынести в отдельный контроллер AirlinesController, иначе AppController скоро станет слишком жирным.

dbodunov replied:
Создал SoapController и перенес маппинги в него

IvanSeregin commented on src/main/java/io/delivery/controller/AppController.java:
```diff
@@ -78,15 +81,51 @@ public String noregisration() {
+    @RequestMapping(value = {"/belavia/timetable/{airport}/{type}/{date}"}, method = RequestMethod.GET)
+    public ModelAndView getListOfFlights(@PathVariable("airport") String airport,
+                                         @PathVariable("type") String type,
+                                         @PathVariable("date") String date) throws IOException, SOAPException {
```
Проверить дату на валидность и вывести сообщение об этом пользователю, вместо вываливания в exception.

dbodunov replied:
В исключение уже не валится. Над валидностью пока работаю.

Dimian13 committed:
added link to Belavia soap description

Dimian13 committed:
changed ClientBelavia to ClientBelaviaAirlines;
added SoapController;
removed mapping from AppConfig and added to SoapController for ClientBelaviaAirlines;
refactored date validation for parameter from mapping to ClientBelaviaAirlines
